lib/order_repository.rb

```diff
@@ -0,0 +1,83 @@
+require_relative './order'
+class OrderRepository
+  def all
+    sql = 'SELECT id, customer_name, placed_date FROM orders;'
+    result_set = DatabaseConnection.exec_params(sql,[])
+
+    orders = []
+
+    result_set.each do |record|
+      order = Order.new
+      order.id = record['id'].to_i
+      order.customer_name = record['customer_name']
+      order.placed_date = record['placed_date']
+
+      items_sql = 'SELECT items.name, items_orders.item_count FROM items JOIN items_orders ON items_orders.item_id = items.id 
+        JOIN orders ON orders.id = items_orders.order_id
+        WHERE orders.id = $1;'
+      items_result = DatabaseConnection.exec_params(items_sql,[record['id']])  
+
+      items_result.each do |record|
+        item = [record['name'],record['item_count'].to_i]
+        order.items << item
+      end
+
+      orders << order
+    end
+    return orders
+  end
+
+  def create(order)
+    DatabaseConnection.connection.transaction do
+      # Create new order in database
+      order_id = self.new_order(order)
+      # Link items to orders and decrease quantity in stock
+      order.items.each do |item| # => ["Apple", 2]
+      self.item_to_order(item,order_id)
+      end
+    end
+    return nil
+  end
+
+  private
+  
+  def new_order(order)
+      # Create new order in orders table
+      sql = 'INSERT INTO orders (customer_name, placed_date) VALUES ($1,$2) RETURNING id;'
+      params = [order.customer_name,order.placed_date]
+
+      result_set = DatabaseConnection.exec_params(sql,params)
+
+      order_id = result_set[0]['id'].to_i
+      return order_id
+  end
+
+  def item_to_order(item,order_id)
```

Comment:
item_to_order method is doing quite a lot; understandably with a many to many relationship; is there a way this could be condensed or broken into smaller parts so it's easier to follow? 

Variables are easily understood throughout!